### PR TITLE
GH-36008: [Ruby][Parquet] Add Parquet::ArrowFileReader#each_row_group

### LINKED ISSUE #36008: [Ruby][Parquet] Add Parquet::ArrowFileReader#each_row_group
### Describe the enhancement requested

It's a convenient method:

```ruby
class Parquet::ArrowFileReader
  def each_row_group
    return to_enum(__method__) {n_row_groups} unless block_given?
    
    n_row_groups.times do |i|
      yield(read_row_group(i))
    end
  end
end
```

### Component(s)

Parquet, Ruby

### PR DESCRIPTION
**Pull request by otegami**
### Rationale for this change
This change allows you to read a large Parquet file per row group.
- ref: https://github.com/apache/arrow/issues/36001

### What changes are included in this PR?
- Add Parquet::ArrowFileReader#each_row_group
- Add the related test about it

### Are these changes tested?
Yes
- I don't have confidence about the test. Could you give me a comment?

### Are there any user-facing changes?
No

Close: https://github.com/apache/arrow/issues/36008
* Closes: #36008

### CODE REVIEW

File: ruby/red-parquet/test/test-arrow-file-reader.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    visible_field = Arrow::Field.new("visible", :boolean)
+    @schema = Arrow::Schema.new([visible_field])
+    visible_arrays = [Arrow::BooleanArray.new([true, false])]
+    visible_array = Arrow::ChunkedArray.new(visible_arrays)
+    table = Arrow::Table.new(@schema, [visible_array])
+    @file = Tempfile.open(["red-parquet", ".parquet"])
+    chunk_size = 1
+    writer = Parquet::ArrowFileWriter.new(table.schema, @file.path)
+    writer.write_table(table, chunk_size)
+    writer.close
```
Comment (kou) [inline ruby/red-parquet/test/test-arrow-file-reader.rb:29]: Could you use `Tempfile.create {...}` and `yield` in `setup` to remove a temporary file as soon as possible?

 ```suggestion
    Tempfile.create(["red-parquet", ".parquet"]) do |file|
      @file = file
      chunk_size = 1
      writer = Parquet::ArrowFileWriter.new(table.schema, @file.path)
      writer.write_table(table, chunk_size)
      writer.close
      yield
    end
```

Comment (kou) [inline ruby/red-parquet/test/test-arrow-file-reader.rb:29]: Could you use more Ruby-ish `ArrowFileWriter.open {...}` API?

```suggestion
    Parquet::ArrowFileWriter.open(table.schema, @file.path) do |writer|
      chunk_size = 1
      writer.write_table(table, chunk_size)
    end
```

File: ruby/red-parquet/test/test-arrow-file-reader.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      second_visible_array = Arrow::ChunkedArray.new(second_visible_arrays)
+
+      Arrow::FileInputStream.open(@file.path) do |input|
+        reader = Parquet::ArrowFileReader.new(input)
+        each_row_group = reader.each_row_group
+        assert_equal({
+                       size: 2,
+                       to_a: [
+                         Arrow::Table.new(@schema, [first_visible_array]),
+                         Arrow::Table.new(@schema, [second_visible_array])
```
Comment (kou) [inline ruby/red-parquet/test/test-arrow-file-reader.rb:46]: We can simplify this:

```suggestion
      Arrow::FileInputStream.open(@file.path) do |input|
        reader = Parquet::ArrowFileReader.new(input)
        each_row_group = reader.each_row_group
        assert_equal({
                       size: 2,
                       to_a: [
                         Arrow::Table.new(@schema, [[true]]),
                         Arrow::Table.new(@schema, [[false]])
```

File: ruby/red-parquet/test/test-arrow-file-reader.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestArrowFileReader < Test::Unit::TestCase
+  def setup
+    visible_field = Arrow::Field.new("visible", :boolean)
+    @schema = Arrow::Schema.new([visible_field])
```
Comment (kou) [inline ruby/red-parquet/test/test-arrow-file-reader.rb:21]: We can simplify this:

```suggestion
    @schema = Arrow::Schema.new(visible: :boolean)
```

File: ruby/red-parquet/test/test-arrow-file-reader.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    visible_arrays = [Arrow::BooleanArray.new([true, false])]
+    visible_array = Arrow::ChunkedArray.new(visible_arrays)
+    table = Arrow::Table.new(@schema, [visible_array])
```
Comment (kou) [inline ruby/red-parquet/test/test-arrow-file-reader.rb:24]: we can simplify this:

```suggestion
    table = Arrow::Table.new(@schema, [[true], [false]])
```

**Review by kou — COMMENTED** [review]
Could you also add a test for with block case?

File: ruby/red-parquet/test/test-arrow-file-reader.rb (hunk 1 above)
Comment (otegami) [reply ruby/red-parquet/test/test-arrow-file-reader.rb]: fix: [0659e9c](https://github.com/apache/arrow/pull/36022/commits/0659e9c63cdd690b6fae9dd5c0bfc28139f7801b)

Comment (otegami) [reply ruby/red-parquet/test/test-arrow-file-reader.rb]: fix: [4c228dd](https://github.com/apache/arrow/pull/36022/commits/4c228dd85612dcb8efc785427b1c798a205b5128)

File: ruby/red-parquet/test/test-arrow-file-reader.rb (hunk 4 above)
Comment (otegami) [reply ruby/red-parquet/test/test-arrow-file-reader.rb]: fix: [5c44ad6](https://github.com/apache/arrow/pull/36022/commits/5c44ad6bf9ade1fa26dfbdf0e5c57021c587dfd5)

File: ruby/red-parquet/test/test-arrow-file-reader.rb (hunk 3 above)
Comment (otegami) [reply ruby/red-parquet/test/test-arrow-file-reader.rb]: fix: [5c44ad6](https://github.com/apache/arrow/pull/36022/commits/5c44ad6bf9ade1fa26dfbdf0e5c57021c587dfd5)

File: ruby/red-parquet/test/test-arrow-file-reader.rb (hunk 2 above)
Comment (otegami) [reply ruby/red-parquet/test/test-arrow-file-reader.rb]: fix: [5c44ad6](https://github.com/apache/arrow/pull/36022/commits/5c44ad6bf9ade1fa26dfbdf0e5c57021c587dfd5)

**Comment by otegami** [comment]
@kou 
Thank you for reviewing 🙏
I've fixed all points you commented on.
Fix: [d18ade2](https://github.com/apache/arrow/pull/36022/commits/d18ade28a711f9580f0d414ac68dddc84a81e7fc): added a test case about 'with block'.

**Review by kou — APPROVED** [review]
+1

Thanks!

**Pull request merged**